### PR TITLE
Nicer spec file

### PR DESCRIPTION
**Pull request by t0fik**
Spec file with recommended build dependency list and sub packages for completion functions.

### CODE REVIEW

**Comment by ErikReider** [comment]
The build script wont complete but it's not a issue with this PR. I'll have to solve this soon. Thanks for the improvement :)

If it goes a few days without a review, please ping me :)

File: build-scripts/swaync.rpkg.spec [hunk 1]
```diff
@@ -84,9 +84,9 @@ This package installs Fish completion files for %{name}
 %{_bindir}/swaync-client
 %{_bindir}/swaync
 %license COPYING
-%config %{_sysconfdir}/xdg/swaync/configSchema.json
-%config %{_sysconfdir}/xdg/swaync/config.json
-%config %{_sysconfdir}/xdg/swaync/style.css
+%config(noreplace) %{_sysconfdir}/xdg/swaync/configSchema.json
+%config(noreplace) %{_sysconfdir}/xdg/swaync/config.json
+%config(noreplace) %{_sysconfdir}/xdg/swaync/style.css
```
Comment (ErikReider) [inline build-scripts/swaync.rpkg.spec:94]: Editing these files are generally discouraged, specially when swaync can parse configs in `XDG_CONFIG_HOME`

File: build-scripts/swaync.rpkg.spec [hunk 2]
```diff
@@ -1,34 +1,73 @@
...
+BuildRequires:  scdoc
+BuildRequires:  pkgconfig(gtk+-3.0) >= 3.22
+BuildRequires:  pkgconfig(gtk-layer-shell-0) >= 0.1
+BuildRequires:  pkgconfig(json-glib-1.0) >= 1.0
+BuildRequires:  pkgconfig(libhandy-1) >= 1.4.0
+BuildRequires:  pkgconfig(glib-2.0) >= 2.50
+BuildRequires:  pkgconfig(gobject-introspection-1.0) >= 1.68
+BuildRequires:  pkgconfig(gee-0.8) >= 0.20
+BuildRequires:  pkgconfig(bash-completion)
+BuildRequires:  pkgconfig(fish)
```
Comment (ErikReider) [inline build-scripts/swaync.rpkg.spec:27]: fish and bash-completion shouldn't be required

File: build-scripts/swaync.rpkg.spec [hunk 3]
```diff
@@ -1,34 +1,73 @@
...
 
 %description
 A simple notification daemon with a GTK gui for notifications and the control center
 
+%package bash-completion
+BuildArch:      noarch
+Summary:        Bash completion files for %{name}
+Provides:       %{alt_pkg_name}-bash-completion = %{version}-%{release}
+
+Requires:       bash
```
Comment (ErikReider) [inline build-scripts/swaync.rpkg.spec:43]: Bash, zsh and fish shouldn't be required

**Review by ErikReider — CHANGES_REQUESTED** [review]
Thanks for the fixes. Just some minor nits.

Btw, do you have any good source for learning more about spec files? I'm pretty new to the RPM packaging system :)

File: build-scripts/swaync.rpkg.spec (hunk 2 above)
Comment (t0fik) [reply build-scripts/swaync.rpkg.spec:27]: Those are only build time requirements, not passed as requirement to package.

File: build-scripts/swaync.rpkg.spec (hunk 3 above)
Comment (t0fik) [reply build-scripts/swaync.rpkg.spec:43]: This is requirement for sub package with completion functions. Main package do not have any shell as requirement.

**Comment by t0fik** [comment]
@ErikReider yes, I'm using rpmlint executed on built packages, vanilla configuration from Fedora (included with rpmlint rpm).

**Review by ErikReider — APPROVED** [review]
LGTM! Thanks for teaching me some new cool spec features :)

**Comment by t0fik** [comment]
My pleasure

**Pull request merged**